src/components/ChattyLLM/ChattyLLMInputForm.vue

```diff
@@ -555,46 +604,64 @@ export default {
 			}
 		},
 
-		async pollGenerationTask(taskId) {
+		async pollGenerationTask(taskId, sessionId) {
 			return new Promise((resolve, reject) => {
 				this.pollMessageGenerationTimerId = setInterval(() => {
 					axios.get(
 						getChatURL('/check_generation'),
-						{ params: { taskId, sessionId: this.active.id } },
+						{ params: { taskId, sessionId } },
 					).then(response => {
 						clearInterval(this.pollMessageGenerationTimerId)
-						resolve(response.data)
+						if (sessionId === this.active.id) {
+							resolve(response.data)
+						} else {
+							console.debug('Ignoring received message for session ' + sessionId + ' that is not selected anymore')
+							// should we reject here?
```

Comment:
let's clear the interval at line 610 when the session is not the same. Polling starts on session switch again anyway.